### ISSUE TITLE
Crash in components which use ripple

### ISSUE THREAD

**Issue by dessant** [material-components/material-components-web#1121]
In #1041 a [new method](https://github.com/material-components/material-components-web/pull/1041/files#diff-9189ce16de72e0d6225ab5651af3fedaR33) is introduced for detecting a css variable bug in Edge. `detectEdgePseudoVarBug` creates a style element and manipulates the object returned by the `sheet` property without checking if `sheet` is not `null`.

~~This may be different on Edge, but in Firefox it will crash, because a newly created style element's `sheet` property will return `null`. Tested with `@material/checkbox@0.4.2` and `@material/ripple@0.8.2`.
https://developer.mozilla.org/en-US/docs/Web/API/HTMLStyleElement~~

```
TypeError: sheet is null
Stack trace:
detectEdgePseudoVarBug@moz-extension://e3572a58-f8e7-4ac2-867d-2f5255353c34/src/options/options.bundle.js:363:3
supportsCssVariables@moz-extension://e3572a58-f8e7-4ac2-867d-2f5255353c34/src/options/options.bundle.js:399:30
browserSupportsCssVars@moz-extension://e3572a58-f8e7-4ac2-867d-2f5255353c34/src/options/options.bundle.js:3865:37
get isSupported_@moz-extension://e3572a58-f8e7-4ac2-867d-2f5255353c34/src/options/options.bundle.js:4078:12
init@moz-extension://e3572a58-f8e7-4ac2-867d-2f5255353c34/src/options/options.bundle.js:4157:9
MDCComponent@moz-extension://e3572a58-f8e7-4ac2-867d-2f5255353c34/src/options/options.bundle.js:137:5
MDCRipple@moz-extension://e3572a58-f8e7-4ac2-867d-2f5255353c34/src/options/options.bundle.js:3829:11
initRipple_@moz-extension://e3572a58-f8e7-4ac2-867d-2f5255353c34/src/options/options.bundle.js:3114:12
MDCCheckbox@moz-extension://e3572a58-f8e7-4ac2-867d-2f5255353c34/src/options/options.bundle.js:3082:20
mounted@moz-extension://e3572a58-f8e7-4ac2-867d-2f5255353c34/src/options/options.bundle.js:2985:25
callHook@moz-extension://e3572a58-f8e7-4ac2-867d-2f5255353c34/src/vue.bundle.js:2332:9
insert@moz-extension://e3572a58-f8e7-4ac2-867d-2f5255353c34/src/vue.bundle.js:3088:7
invokeInsertHook@moz-extension://e3572a58-f8e7-4ac2-867d-2f5255353c34/src/vue.bundle.js:4787:9
patch@moz-extension://e3572a58-f8e7-4ac2-867d-2f5255353c34/src/vue.bundle.js:4940:5
[8]/lifecycleMixin/Vue.prototype._update@moz-extension://e3572a58-f8e7-4ac2-867d-2f5255353c34/src/vue.bundle.js:2116:16
updateComponent@moz-extension://e3572a58-f8e7-4ac2-867d-2f5255353c34/src/vue.bundle.js:2219:7
get@moz-extension://e3572a58-f8e7-4ac2-867d-2f5255353c34/src/vue.bundle.js:2534:13
run@moz-extension://e3572a58-f8e7-4ac2-867d-2f5255353c34/src/vue.bundle.js:2604:17
flushSchedulerQueue@moz-extension://e3572a58-f8e7-4ac2-867d-2f5255353c34/src/vue.bundle.js:2392:5
[8]/queueNextTick/<@moz-extension://e3572a58-f8e7-4ac2-867d-2f5255353c34/src/vue.bundle.js:643:11
nextTickHandler@moz-extension://e3572a58-f8e7-4ac2-867d-2f5255353c34/src/vue.bundle.js:588:7
```

**Comment by acdvorak** [comment]
Thanks for filing this issue!

We haven't been able to reproduce the error, but it looks like you're encountering it in a browser extension - is that right? If so, are you able to repro the error _outside_ of the extension?

If it only happens in the extension, it's possible that stylesheets behave differently in extensions than on "regular" web pages (though I could certainly be wrong).

On a normal web page, adding the `<style>` element to the DOM ensures that the `sheet` property returns a non-`null` value, and we're calling `document.head.appendChild(styleNode)` before accessing the `sheet` property, so I'm not sure why this would be happening.

**Comment by dessant** [comment]
Yes, I've stumbled upon this in an extension. After further testing, it seems this is not due to browser differences, but CSP.

I was able to reproduce the issue in both Chrome and Firefox by loading this page locally:
```html
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv='Content-Security-Policy' content="style-src 'self';">
  </head>
  <body>
    <script>
    var styleNode = document.createElement('style');
    document.head.appendChild(styleNode);
    console.log(styleNode.sheet);
</script>
  </body>
</html>
```
```
options.html:10 Refused to apply inline style because it violates the following Content Security Policy directive: "style-src 'self'". Either the 'unsafe-inline' keyword, a hash ('sha256-47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU='), or a nonce ('nonce-...') is required to enable inline execution.
options.html:11 null
```

**Comment by nealedj** [comment]
After updating from 0.16.0 to 0.19.0 I'm now seeing this issue when using `MDCDialog`.

> Refused to apply inline style because it violates the following Content Security Policy directive: "style-src 'self' ... 'nonce-xxxxx'". Either the 'unsafe-inline' keyword, a hash ('sha256-xxxxx'), or a nonce ('nonce-...') is required to enable inline execution.

> material-components-web.js:1047 Uncaught TypeError: Cannot read property 'insertRule' of null
>     at detectEdgePseudoVarBug (material-components-web.js:1047)
>     at Object.supportsCssVariables (material-components-web.js:1083)
>     at Object.browserSupportsCssVars (material-components-web.js:600)
>     at MDCRippleFoundation.get (material-components-web.js:2480)
>     at MDCRippleFoundation.init (material-components-web.js:2618)
>     at MDCRipple.MDCComponent (material-components-web.js:287)
>     at new MDCRipple (material-components-web.js:508)
>     at MDCDialog.initialize (material-components-web.js:3673)
>     at MDCDialog.MDCComponent (material-components-web.js:282)
>     at new MDCDialog (material-components-web.js:3662)

**Comment by nealedj** [comment]
Quick update as it's causing a pretty significant headache for me right now.

This appears to affect any components which make use of MDCRipple. So all dialogs are currently broken.

I've tested this in Chrome 60.0.3112.113 and FF 55.0.3. Both fail at this point.

I actually updated the library from 0.15.0 to 0.19.0 due to a separate error (#1196). Since that's still an issue I'm currently unable to deploy my project.
